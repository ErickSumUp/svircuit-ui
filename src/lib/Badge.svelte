<script lang="ts">
	/**
	 * Choose from 4 style variants. Default: 'neutral'
	 */
	export let variant: 'neutral' | 'success' | 'warning' | 'danger' | 'promo' = 'neutral';
	/**
	 * Use the circular badge to indicate a count of items related to an element.
	 */
	export let circle = false;
</script>

<span
	class="base"
	class:neutral={variant === 'neutral'}
	class:success={variant === 'success'}
	class:warning={variant === 'warning'}
	class:danger={variant === 'danger'}
	class:promo={variant === 'promo'}
	class:circle
>
	<slot />
</span>

<style>
	.base {
		display: inline-block;
		padding: 2px var(--cui-spacings-byte);
		font-size: var(--cui-typography-body-two-font-size);
		font-weight: var(--cui-font-weight-bold);
		line-height: var(--cui-typography-body-two-line-height);
		text-align: center;
		letter-spacing: 0.25px;
		border-radius: var(--cui-border-radius-pill);
	}
	.circle {
		display: flex;
		align-items: center;
		justify-content: center;
		width: 24px;
		height: 24px;
		padding: 2px 4px;
	}
	.success {
		color: var(--cui-fg-on-strong);
		background-color: var(--cui-bg-success-strong);
	}
	.warning {
		color: var(--cui-fg-on-strong);
		background-color: var(--cui-bg-warning-strong);
	}
	.danger {
		color: var(--cui-fg-on-strong);
		background-color: var(--cui-bg-danger-strong);
	}
	.neutral {
		color: var(--cui-fg-normal);
		background-color: var(--cui-bg-highlight);
	}
	.promo {
		color: var(--cui-fg-on-strong);
		background-color: var(--cui-bg-promo-strong);
	}
</style>
