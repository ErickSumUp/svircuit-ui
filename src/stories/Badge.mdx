import * as Stories from './Badge.stories.svelte';
import { Canvas, Controls, Meta } from '@storybook/blocks';

<Meta of={Stories} />

# Badge

Badges are non-actionable elements in the UI, used to indicate the status of an object.

<Canvas withToolbar={true} of={Stories.Base} />
<Controls />

## Best practices

- Use our colour patterns so that users can learn and identify status at a glance.
- Create a clear label with a short, scannable text.
- Position a badge so that it is obvious what content the status is related to.

## Usage guidelines

- **Do** use badges to indicate the status of data rows in a table.
- **Do** use badges to bring attention to a specific area on the screen (like a new feature).
- **Do not** use badges to indicate default statuses (e.g. in the Transaction History, a transaction should only include a status if it's anything other than default; failed, refunded, paid out, etc.)

## Component variations

### Variants

The badges come in a number of variants based on their semantic meaning.

<Canvas of={Stories.Variants} />

### Circular badges

Badges that receive the `circle` prop can be used to indicate notifications.

<Canvas of={Stories.Circular} />
