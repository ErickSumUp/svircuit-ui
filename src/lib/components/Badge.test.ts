import { render } from '@testing-library/svelte';
import '@testing-library/jest-dom';
import Badge from '$lib/components/Badge.svelte';
import { test, expect } from 'vitest';

test('should match story case variants', async () => {
  const { container } = render(Badge, {
    props: { variant: 'neutral' }
  });
  expect(container).toMatchSnapshot();
});

test('should match story case circular', async () => {
  const { container } = render(Badge, {
    props: { circle: true, variant: 'neutral' }
  });
  expect(container).toMatchSnapshot();
});
