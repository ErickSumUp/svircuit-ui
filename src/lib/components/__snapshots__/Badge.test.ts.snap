// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match story case 1`] = `
<div>
  <span
    class="base neutral svelte-xka8p6"
  />
</div>
`;

exports[`should match story case circular 1`] = `
<div>
  <span
    circular="true"
    class="base neutral svelte-xka8p6"
  />
</div>
`;
